build.rs

```diff
@@ -12,12 +12,43 @@
 use std::{env, path::Path};
 
 fn main() {
-    let target_os = env::var_os("CARGO_CFG_TARGET_OS").unwrap();
+    if let Ok(handler) = env::var("DEP_SENTRY_NATIVE_HANDLER") {
+        if cfg!(feature = "copy-handler") {
+            let out_dir = env::var("OUT_DIR").expect("out dir not set");
 
-    if target_os == "windows" || target_os == "macos" {
-        println!(
-            "cargo:rustc-env=HANDLER={}",
-            AsRef::<Path>::as_ref(&env::var_os("DEP_SENTRY_NATIVE_HANDLER").unwrap()).display()
-        );
+            // OUT_DIR will point to a directory unique to each crate, which be something like
+            // target/debug/build/sentry-contrib-native-sys-f734ae671f48a2d5/out, so we go up
+            // 3 parents to get to the root directory (target/debug in this case), which is
+            // where the final binary artifacts will be placed by cargo, so we copy the
+            // crashpad_handler to the same directory to fit with the default expectation the
+            // handler is next to the executable it is monitoring, and so that scripts/programs
+            // that want to package the crashpad_handler along with the executable don't have
+            // to trawl through the target directory looking for it, as, AFAICT, there is no
+            // convenient way to specify the output path of the handler where it is available
+            // to eg other builds scripts, as noted by cargo
+            //
+            // > Note that metadata is only passed to immediate dependents, not transitive dependents.
+            //
+            // And we can't assume that this crate will be a direct dependency of the crate.
+            let out_dir = Path::new(&out_dir);
+            let bin_dir = out_dir
+                .parent()
+                .unwrap()
+                .parent()
+                .unwrap()
+                .parent()
+                .unwrap();
+
+            let handler = Path::new(&handler);
+            let bin_path = bin_dir.join(
+                handler
+                    .file_name()
+                    .expect("handler doesn't have a file name"),
+            );
+
+            std::fs::copy(&handler, &bin_path).expect("failed to copy sentry crash handler");
+        }
+
+        println!("cargo:rustc-env=SENTRY_HANDLER={}", handler);
```

Comment:
Is there a reason why you renamed this from `HANDLER` to `SENTRY_HANDLER`? It's only local, other crates can't access this environment variable.

Reply:
I think the environment is available in all dependent crates, it's only the `DEP_` ones generated by cargo that are only visible to the immediate dependencies and not transitive ones....I think? `HANDLER` just seemed too generic to me, but I don't think it matters much.

Reply:
I just double-checked, it's not visible anywhere else except the crate itself, crates depending on it can't see it.
In any case, you are right about it being too generic.
> ... but I don't think it matters much.

Yes 😄, thanks for the input though!